### DISCUSSION TITLE
Why does meow-mark-word not move in the current direction?

### DISCUSSION THREAD

**Discussion by Stebalien** [meow-edit/meow#387] [Q&A]
`meow-line` marks the current line, and then the next and the next. But `meow-mark-word` doesn't.

It's not a huge issue, just oddly inconsistent so I was wondering if there was some reason for this behavior.

**Comment by nchisholm** [comment]
I think this would be potentially useful behavior.  Reversed selections could also select words backwards, also analogous to line selection.   I think it wouldn't be too hard to define a function `+meow-mark-or-expand-word` that would call `meow-mark-word`, `meow-forward-word`, or `meow-backward-word` depending on whether the mark is active and the direction of the selection.

Due to some replication of `meow-forward-word` and `meow-backward-word`, this might be useful in combination with a function that shrank selections one word at a time, so you could, e.g., select to the end of a sentence and then "subtract" a couple of words from the end.  Currently, you can only expand selections.

**Reply by DogLooksGood** [reply]
To subtract words, you can use builtin movements, e.g. `M-f`, `M-b`.

**Reply by nchisholm** [reply]
Right! Silly I did not think of that. Would there be any way to also couple M-f and M-b to Meow's repeat-count hints?

**Reply by DogLooksGood** [reply]
No, there isn't a way to do that, because they are just built-in commands. It's better to create the selection directly by describing where you want to go(e.g. end of a symbol, point after a specific character), instead of constructing the path(e.g. line * 3 + word *2, symbol * 3 - word).  IMO, the previous way is much easier, and reasonable.  Unless you want a very large selection, in this case you can expand with built-in commands.
